### PR TITLE
Expose cookies as esm

### PR DESCRIPTION
**Pull request by mattcosta7**
Expose cookies esm build.

Added tsup, and config similar to other msw related libraries

Once this was done, there were some bundle related issues, since set-cookie-parser is only released as commonjs.

In order to support this, we bundle that library by moving it into devDependencies instead of declaring it as a dependency. By default, tsup treats all 'dependencies' as external in an esm build, so this is not necessarily standard, but works to ensure that this library is treated as esm compatible.

### CODE REVIEW

File: esm/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (kettanaito) [inline esm/package.json:1]: If the library paths is mapped on `.` for ESM, is there a need to create the `/esm` directory?

The purpose we have these dirs in `msw` is to enable imports like `msw/node` where the structure is:

```
msw/
  node/
    package.json
```

But that's not the case with ESM in this package, is it? Or is it intended to be imported like `@mswjs/cookies/esm`? 🤔

Comment (kettanaito) [inline esm/package.json:1]: Is this `/esm` directory necessary if we map imports to root `.`?

Comment (mattcosta7) [reply esm/package.json:1]: We could probably avoid it. I'll check locally when I get some time and remove if that works as well

**Comment by mattcosta7** [comment]
I'll take a look at ci here when I get some time (probably this weekend)

**Comment by kettanaito** [comment]
Huge thanks to you for this titanic work! 👏

**Pull request closed without merging**